x-pack/platform/plugins/shared/onechat/public/application/components/tools/execute/test_tools.tsx

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiButton,
+  EuiFieldText,
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiForm,
+  EuiFormRow,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { ToolDefinition } from '@kbn/onechat-common';
+import { i18n } from '@kbn/i18n';
+import { CodeEditor } from '@kbn/code-editor';
+import { useExecuteTool } from '../../../hooks/tools/use_execute_tools';
+import type { ExecuteToolResponse } from '../../../../../common/http_api/tools';
+
+interface OnechatTestToolFlyoutProps {
+  isOpen: boolean;
+  isLoading?: boolean;
+  tool?: ToolDefinition;
+  onClose: () => void;
+}
+
+const getParameters = (tool: ToolDefinition | undefined) => {
+  if (!tool) return [];
+
+  const fields: Array<{ name: string; label: string; value: string; type: string }> = [];
+  if (tool.configuration && tool.configuration.params) {
+    const params = tool.configuration.params as Record<string, any>;
+    Object.entries(params).forEach(([paramName, paramConfig]) => {
+      fields.push({
+        name: `${paramName}`,
+        label: `${paramName}`,
+        value: '',
+        type: paramConfig.type || 'text',
+      });
+    });
+  }
+
+  return fields;
+};
+
+export const OnechatTestFlyout: React.FC<OnechatTestToolFlyoutProps> = ({
+  isOpen,
+  isLoading,
+  tool,
+  onClose,
+}) => {
+  const [formData, setFormData] = useState<Record<string, any>>({});
```

Comment:
Let's use react hook form rather than this generic type. We can infer what fields we have from tool schema. 

React hook form should provide validation (e.g. what fields are required or what are optional)

e.g. form doesn't prevent me to submit form with a bad state (empty input)

<img width="812" height="290" alt="Image" src="https://github.com/user-attachments/assets/7b2114c4-6eff-4676-bed1-f86dd32573bf" />